internal/ottlfuncs/func_is_match_ruby_regex.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ottlfuncs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	rubex "github.com/GoogleCloudPlatform/opentelemetry-operations-collector/third_party/go-oniguruma"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type IsMatchRubyRegexArguments[K any] struct {
+	Target  ottl.StringLikeGetter[K]
+	Pattern string
+}
+
+func NewIsMatchRubyRegexFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("IsMatchRubyRegex", &IsMatchRubyRegexArguments[K]{}, createIsMatchRubyRegexFunction[K])
```

Comment:
This line can move into `LogFunctions()` and you don't need this as a separate function at all